src/main/client/app/shared/components/sidebar/side-bar.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+  <div :class="collapsed ? 'sidebar collapsed' : 'sidebar'">
+    <div class="sidebar-header">
+      <app-side-bar-toggle
+        v-model="collapsed"
+        @input="toggleSidebar"
+      />
+    </div>
+    <div class="sidebar-body">
+      <app-side-bar-links :collapse="collapsed" />
+    </div>
+    <div
+      v-if="buildInfo"
+      class="sidebar-footer"
+    >
+      <app-side-bar-infos :build-info="buildInfo" />
+    </div>
+  </div>
+</template>
+
+<script>
+  import AppSideBarToggle from '@/shared/components/sidebar/side-bar-toggle.vue';
+  import AppSideBarLinks from '@/shared/components/sidebar/side-bar-links.vue';
+  import AppSideBarInfos from '@/shared/components/sidebar/side-bar-infos.vue';
+  import {
+    getCookie,
+    setCookie,
+  } from '@/shared/services';
+  import { getBuildInfo } from '@/shared/api';
+
+  const SIDEBAR_COOKIE = 'sidebar_collapsed';
+  const SIDEBAR_MAX_AGE = 60 * 60 * 24 * 365 * 10;
+  const WINDOW_MIN_SIZE = 1199;
```

Comment:
Why these magical value ? 🧙

Reply:
it matches the max-value for the responsive css in the default layout:

```css
  @media (max-width: 1199px) {
    .content {
      padding-left: 95px;
    }
  }
```

Reply:
no idea to link the two of them :confused: